pkg/flow/flow.go

```diff
@@ -145,13 +145,27 @@ type Flow struct {
 
 // New creates a new, unstarted Flow controller. Call Run to run the controller.
 func New(o Options) *Flow {
-	return newController(newModuleRegistry(), o)
+	return newController(controllerOptions{
+		Options:        o,
+		ModuleRegistry: newModuleRegistry(),
+		IsModule:       false, // We are creating a new root controller.
+	})
+}
+
+// controllerOptions are internal options used to create both root Flow
+// controller and controllers for modules.
+type controllerOptions struct {
+	Options
+
+	ComponentRegistry controller.ComponentRegistry // Custom component registry used in tests.
+	ModuleRegistry    *moduleRegistry              // Where to register created modules.
+	IsModule          bool                         // Whether this controller is for a module.
```

Comment:
Minor personal preference: I'd prefer if this is an enum which indicates what the controller is for. It'd answer the question of "what is this, if it's not a module controller?". E.g. it'd be clear that this is either a `ControllerModule` vs `ControllerRoot`). But I'm ok with this staying as it is.